packages/fxa-auth-server/lib/routes/totp.js

```diff
@@ -519,9 +535,14 @@ module.exports = (
 
         await customs.checkAuthenticated(request, uid, email, 'verifyTotpCode');
 
-        const token = await db.totpToken(sessionToken.uid);
-        const sharedSecret = token.sharedSecret;
-        const tokenVerified = token.verified;
+        // const token = await db.totpToken(sessionToken.uid);
+        // const sharedSecret = token.sharedSecret;
+        // const tokenVerified = token.verified;
+        let sharedSecret = await authServerCacheRedis.get(`totp:secret:${uid}`);
+        let tokenVerified = false;
+        if (sharedSecret == null) {
+          ({ sharedSecret, verified: tokenVerified } = await db.totpToken(uid));
+        }
```

Comment:
We should check if the account has totp enabled first since that'd be the use case far more frequently.